collaboration-service/test/test.js

```diff
@@ -0,0 +1,23 @@
+import io from "socket.io-client"
+import chai from 'chai'
+import httpServer from '../index.js'
+import { roomId, codeChange, localUrl } from './testData.js'
+let socketUrl = localUrl;
+
+chai.should();
+
+describe("Code change", () => {
+    it("client2 should receive the change in code", done => {
+        const client1 = io.connect(socketUrl);
+        const client2 = io.connect(socketUrl)
+        client1.emit("join", roomId);
+        client2.emit("join", roomId);
+        client1.emit("change", codeChange)
+        client2.on("receive", (payload) => {
+            if (payload.code) {
+                payload.code.should.equal("print('Hello World')");
+                done();
+            }
+        })
+    })
+})
```

Comment:
Should add Collab Service CI along with tests, maybe can separate testing with CI in another PR

Reply:
Yes can separate tests + CI in another PR